### PR TITLE
GEODE-6327: Ability to specify identity fields on JSON documents

### PR DESCRIPTION
**Pull request by mcmellawatt**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [X] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [X] Does `gradlew build` run cleanly?

- [X] Have you written or updated unit tests to verify your changes?

- [N/A] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/integrationTest/java/org/apache/geode/pdx/JSONFormatterBasicJUnitTest.java [hunk 1]
```diff
@@ -99,7 +105,62 @@ public void emptyObjectParses() {
...
+      assertThat(pdxInstance.isIdentityField(identityField)).isTrue();
+      assertThat(pdxInstance.isIdentityField(nonIdentityField)).isFalse();
+      assertThat(pdxInstance.isIdentityField(nonExistentField)).isFalse();
+      assertThat(pdxInstance.hasField(identityField)).isTrue();
+      assertThat(pdxInstance.hasField(nonIdentityField)).isTrue();
+      assertThat(pdxInstance.hasField(nonExistentField)).isFalse();
+      assertThat(deserializedJsonObjectString).isEqualTo(jsonObjectString);
+    } finally {
+      if (originalSortJsonFieldNamesProperty != null) {
+        System.setProperty(JSONFormatter.SORT_JSON_FIELD_NAMES_PROPERTY,
```
Comment (bschuchardt) [inline geode-core/src/integrationTest/java/org/apache/geode/pdx/JSONFormatterBasicJUnitTest.java]: You should use the RestoreSystemProperties rule.  It'll make sure things are put back the way they should be.  As it stands it looks like you'll leave this property set if it wasn't set to something else before this test is run.

Comment (mcmellawatt) [reply geode-core/src/integrationTest/java/org/apache/geode/pdx/JSONFormatterBasicJUnitTest.java]: Thanks Bruce - will do.

**Comment by PurelyApplied** [comment]
Please do not merge a PR with a failing precheckin without commenting on the nature of the failing test, such as if there is an existing ticket or known issue.  Without some external knowledge, it seems as though this merge should be reverted until that test failure can be resolved.

**Comment by mcmellawatt** [comment]
@PurelyApplied This failure is form of known issue GEODE-6326 (https://issues.apache.org/jira/browse/GEODE-6326).  I did not know it was policy to comment on the PR if a known unrelated failure occurs.  The failure is related to eventual consistency when concurrent events are applied to a region.  This PR adds the ability to specify identity fields using JSON formatter, which is a completely unrelated part of the code base.  Still, I'm happy to comment on future PRs with the known ticket number if that is now our policy.  I have also retriggered the PR pipeline in previous PRs, and can do that as well, but I'm not sure if its worth the resources just to prove that a known issue is known/flakey.

**Comment by PurelyApplied** [comment]
Oh, I don't know that we have an official policy.  I was just trying to say that, if you know what the issue is (which I assume you do since you'll have investigated your test failures), then I personally appreciate having those breadcrumbs.

Although maybe we should have a policy... I should probably send something out to the dev list about it.

**Pull request merged**